File: app_backend/app/models/vessel.py
from sqlalchemy import Column, Integer, String,Float
from ..database import Base
from sqlalchemy.orm import relationship


class Vessel(Base):
    __tablename__ = "vessels"

    id = Column(Integer, primary_key=True, index=True)
    name = Column(String, nullable=False)
    registration_number = Column(String, unique=True, nullable=False)
    capacity = Column(Integer, nullable=True)
    tonnage = Column(Float, nullable=True)

    documents = relationship("Document", back_populates="vessel", cascade="all, delete")

File: pesca_app/lib/models/vessel.dart
class Vessel {
  final int id;
  final String name;
  final String registrationNumber;
  final double capacity;
  final double tonnage;

  Vessel({
    required this.id,
    required this.name,
    required this.registrationNumber,
    required this.capacity,
    required this.tonnage,
  });

  factory Vessel.fromJson(Map<String, dynamic> json) {
    return Vessel(
      id: json['id'],
      name: json['name'],
      registrationNumber: json['registration_number'],
      capacity: json['capacity'].toDouble(),
      tonnage: json['tonnage'].toDouble(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'registration_number': registrationNumber,
      'capacity': capacity,
      'tonnage': tonnage,
    };
  }
}